server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueFactory.java

```diff
@@ -58,19 +63,28 @@ public RabbitMQMailQueue createQueue(String name) {
     @Override
     public Set<RabbitMQMailQueue> listCreatedMailQueues() {
         return mqManagementApi.listCreatedMailQueueNames()
-            .map(mailQueueFactory::create)
-            .collect(Guavate.toImmutableSet());
+            .map(this::getOrElseCreateLocally)
+            .collect(ImmutableSet.toImmutableSet());
     }
 
     private RabbitMQMailQueue attemptQueueCreation(MailQueueName mailQueueName) {
         rabbitClient.attemptQueueCreation(mailQueueName);
-        return mailQueueFactory.create(mailQueueName);
+        return createLocally(mailQueueName);
     }
 
     private Optional<RabbitMQMailQueue> getQueue(MailQueueName name) {
         return mqManagementApi.listCreatedMailQueueNames()
             .filter(name::equals)
-            .map(mailQueueFactory::create)
+            .map(this::getOrElseCreateLocally)
             .findFirst();
     }
+
+    private RabbitMQMailQueue getOrElseCreateLocally(MailQueueName name) {
+        return Optional.ofNullable(mailQueueStoreLocal.get(name))
+            .orElseGet(() -> createLocally(name));
+    }
+
+    private RabbitMQMailQueue createLocally(MailQueueName name) {
```

Comment:
I would argue that `getOrElseCreateLocally` is a pure duplicate of  `createLocally` :-)